pypesto/collections/collections.py

```diff
@@ -0,0 +1,484 @@
+import numpy as np
+import pandas as pd
+from typing import Sequence, Tuple, Callable, Dict
+
+from ..prediction import PredictionResult, PredictionConditionResult
+from .constants import (PREDICTOR, PREDICTION_ID, PREDICTION_RESULTS,
+                        PREDICTION_ARRAYS, PREDICTION_SUMMARY, OUTPUT,
+                        OUTPUT_SENSI, TIMEPOINTS, X_VECTOR, NX, X_NAMES,
+                        NVECTORS, VECTOR_TAGS, PREDICTIONS, MODE_FUN,
+                        CollectionType, COLLECTION_TYPE, MEAN, MEDIAN,
+                        STANDARD_DEVIATION, PERCENTILE, SUMMARY, LOWER_BOUND,
+                        UPPER_BOUND)
+
+
+class CollectionPrediction:
```

Comment:
Any reason to call this class/file "collection" instead of "ensemble"?

Reply:
The idea was to have a term covering "ensemble" and "sample" ... ;)

Reply:
For me, an "ensemble" is something that explicitly doesn't attempt to be representative, at least that's the way it way recently used in literature... So I didn't want to go for "ensemble", as a lot of the stuff can be reused 1:1 for samples...
However: no big issue to just rename this all to "ensemble", if that less misleading...

Reply:
Hmm, I would still call parameter sets from sampling an "ensemble". Looking at https://en.wikipedia.org/wiki/Ensemble_learning they list Bayesian model averaging also as ensemble method. Similarly https://en.wikipedia.org/wiki/Statistical_ensemble_(mathematical_physics) includes Monte Carlo samplings. Accordingly, I would say that samples are also ensembles, but not the other way around so calling this "Ensemble" is probably more intuitive for most. Happy to hear other opinions though.

Reply:
> samples are also ensembles, but not the other way around so calling this "Ensemble" is probably more intuitive for most

I consider myself as part of *most* :)

Reply:
No, totally fine. I was really struggling with a good naming scheme. If literature also lists samples as ensembles, it really makes sense to adapt this. I'll do so.